### DISCUSSION TITLE
Attempted to perform an unauthorized operation. Get-PnPSiteCollectionAdmin:

### DISCUSSION THREAD

**Discussion by Lpt-wq** [pnp/powershell#846] [General]
I am attempting to get a list of all site owners and members of the sites in my environment using PowerShell. i dont' find a way to do this so I was trying to loop through sites using connect-pnponline then running get-pnpsitecollectionadmin, but I am getting the error " Attempted to perform an unauthorized operation."

I am the sharepoint admin of the current environment. I am using pnp.powershell v 1.2.11

is there a way to get all owners and members of all sites in the current environment? do I need to be an admin to each site to be able to access its info in powershell?

**Comment by gautamdsheth** [comment]
Moving this to discussion as this is not an issue with the PnP PowerShell itself.

**Reply by Lpt-wq** [reply]
Thanks @gautamdsheth , but why is it giving me an error? shouldn't pnp powershell be executed as an admin in the environment if i am using sharepoint admin account?

**Reply by AndersRask** [reply]
Remember that you need to connect to each site individually when you call Get-PnPSiteCollectionAdmin.
Below is some untested code that i guess would work:

```
$adminConnection = Connect-PnPAdmin -ReturnConnection <admin>
$cred = Get-Credential
Get-PnPTenantSite -Connection $adminConnection | % {
try {
    $siteConn = Connect-PnPOnline -ReturnConnection -Url $_.Url -Credentials $cred
    Get-PnPSiteCollectionAdmin -Connection $siteConn
} finally {
    Disconnect-PnPOnline -Connection $siteConn
}
```

**Reply by azureskydiver** [reply]
`Connect-PnPAdmin` doesn't seem to exist (anymore).
